contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+                INonfungiblePositionManager.IncreaseLiquidityParams({
+                    tokenId: tokenId,
+                    amount0Desired: WETHBalance,
+                    amount1Desired: OETHbRequired,
+                    // we don't need slippage protection checking for that in _checkLiquidityWithinExpectedShare
+                    amount0Min: 0,
+                    // we don't need slippage protection checking for that in _checkLiquidityWithinExpectedShare
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
+
+            updateUnderlyingAssets();
+            emit LiquidityAdded(
+                WETHBalance, // WETHAmountDesired
+                OETHbRequired, // OETHbAmountDesired
+                WETHAmountSupplied, // WETHAmountSupplied
+                OETHbAmountSupplied, // OETHbAmountSupplied
+                tokenId, // tokenId
+                underlyingAssets
+            );
+        }
+
+        // burn remaining OETHb
+        _burnOETHbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    /**
+     * @dev Check that the liquidity in the pool is withing the expected WETH to OETHb ratio
+     */
+    function _checkLiquidityWithinExpectedShare() internal {
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        require(WETHPositionBalance + OETHbPositionBalance > 0, "Can not withdraw full position");
+
+        uint160 currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(currentPrice >= sqrtRatioX96Tick0 && currentPrice <= sqrtRatioX96Tick1, "Not in expected tick range");
+
+        uint256 currentWethShare = 0;
+        if (WETHPositionBalance != 0) {
+            currentWethShare = WETHPositionBalance.divPrecisely(WETHPositionBalance + OETHbPositionBalance);
+        }
+
+        if (currentWethShare < poolWethShareVarianceAllowed || // uint256 mustn't go below 0  
+            poolWethShare < currentWethShare - poolWethShareVarianceAllowed ||
+            poolWethShare > currentWethShare + poolWethShareVarianceAllowed) {
+
+            revert PoolRebalanceOutOfBounds(
+                currentWethShare,
+                poolWethShare,
+                WETHPositionBalance,
+                OETHbPositionBalance
+            );
+        } else {
+            emit PoolRebalanced(
+                currentWethShare,
+                poolWethShare,
+                WETHPositionBalance,
+                OETHbPositionBalance
+            );
+        }
+    }
+
+    /**
+     * Burns any lingering OETHb tokens still remaining on the contract
+     */
+    function _burnOETHbOnTheContract() internal {
+        uint256 OETHbBalance = IERC20(OETHb).balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(OETHbBalance);
+    }
+
+    function updateUnderlyingAssets() internal {
```

Comment:
This is an internal function but doesn't have `_` in its name

Reply:
yes, agreed: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/56bb9188bd55b64fdaa7e0db6bae1dddd3d30293